cadquery/assembly.py

```diff
@@ -106,16 +109,36 @@ def _getAxis(self, arg: Shape) -> Vector:
 
         return rv
 
-    def _getPlane(self, arg: Shape) -> Plane:
+    def _getPln(self, arg: Shape) -> gp_Pln:
 
         if isinstance(arg, Face):
-            normal = arg.normalAt()
+            rv = gp_Pln(self._getPnt(arg), arg.normalAt().toDir())
         elif isinstance(arg, (Edge, Wire)):
             normal = arg.normal()
+            origin = arg.Center()
+            plane = Plane(origin, normal=normal)
+            rv = plane.toPln()
         else:
-            raise ValueError(f"Can not get normal from {arg}.")
-        origin = arg.Center()
-        return Plane(origin, normal=normal)
+            raise ValueError(f"Can not construct a plane for {arg}.")
+
+        return rv
+
+    def _getPnt(self, arg: Shape) -> gp_Pnt:
+
+        # check for infinite face
+        if isinstance(arg, Face) and any(
+            Precision.IsInfinite_s(x) for x in BRepTools.UVBounds_s(arg.wrapped)
+        ):
+            # fall back to gp_Pln center
+            pln = arg.toPln()
+            center = Vector(pln.Location())
+        else:
+            center = arg.Center()
+
+        if center.Length > 1e98:
```

Comment:
Do you still need this given the check at 129?

Reply:
I don't think so, but I'm not certain. Can any other `cq.Shape` have a center at 1e99?

Shells do not (I think):
```python
>>> f = cq.Face.makePlane(basePnt=cq.Vector(10, 10, 10), dir=cq.Vector(1, 1, 1))
>>> f.Center()
Vector: (5e+99, -5e+99, 5e+99)
>>> cq.Shell.makeShell([f]).Center()
Vector: (0.0, 0.0, 0.0)
>>> 
```

I'm going to assume no other Shape can have it's center at 1e99 and remove this check, but let me know if that's not the case.

Reply:
Looks good